### DISCUSSION TITLE
Transition to a Zero-Cost Development Solution for OpenFEDEM

### DISCUSSION THREAD

**Discussion by oeysstra** [openfedem/openfedem.org#4] [General]
Currently, the development of OpenFEDEM is conducted using Visual Studio and the Intel Compiler. While these tools offer robust capabilities, their cost is a significant consideration for our open-source project. To ensure that OpenFEDEM remains accessible and sustainable, we are exploring zero-cost development environments and compilers that can replace our current setup.

We invite the community to participate in this discussion to help identify and evaluate alternative tools. Potential alternatives include:

IDE Options:
- Visual Studio Code
- Eclipse
- Code::Blocks
- ...

Compiler Options:
- GCC (GNU Compiler Collection)
- Clang/LLVM
- MinGW (Minimalist GNU for Windows)
- ...

Our goal is to ensure a seamless transition that maintains the high performance and reliability standards of OpenFEDEM. Your insights, experiences, and recommendations are invaluable to us. Let's collaborate to find the best solutions that support our commitment to open-source principles and cost-effectiveness.

**Comment by freeNastran** [comment]
Visual Studio express and the Intel Compiler are free. But I'd like to compile the solver with gcc/gfortran under Windows 10.

**Reply by kmokstad** [reply]
I'm not aware that the Intel Fortran Compiler is available for free. But if that is true, I like to know more about it. Using gcc under WIndows should be relatively straight forward. We have built and tested all solver codes with gcc/gfortran on Linux, so I assume doing the same for Windows (using MinGW) should work fine as well. It may only need some adjustments in some CMakeLists.txt files. The GUI code has also been built with gcc under Linux (Ubuntu 22.04) but not tested that much yet.

**Reply by kmokstad** [reply]
It turns out that the Intel compilers have been freely available for a while (unless you need support), see, e.g., the post on [reddit.com](https://www.reddit.com/r/fortran/comments/1900xtq/intel_fortran_on_visual_studio_market/?rdt=34557). You can download the needed packages from this [Intel site](https://www.intel.com/content/www/us/en/developer/articles/tool/oneapi-standalone-components.html). I tried that now and integrated with Visual Studio 2022, and everything builds fine. Remember to also install the MKL package before building the Fedem solvers.

With this, the need for alternative Fortran compilers on Windows is less imminent. However, anyone who wants to explore using gcc/gfortran instead on Windows are welcome to do so, and share the experiences with the community.

**Comment by kmokstad** [comment]
There is also an option to use Microsoft's msbuild tool to perform the Windows build without the need for the full-blown Visual Studio IDE. I think @gglan already successfully tried this.
